### ISSUE TITLE
Upgrade path for people presently using spawnfcgi + munin-cgi-* 

### ISSUE THREAD

**Issue by kentfredric** [munin-monitoring/munin#538]
Its not abundantly clear to me how people use the new development munin underneath an existing web server.

http://guide.munin-monitoring.org/en/latest/develop/environment.html

Describes how to run a self-hosting munin HTTP server, but this is probably not desirable for people who are currently using the FCGI interface.

Neither will it be satisfactory to run the new scripts as plain CGI-bin scripts.

I've scraped various documents about HTTP::Server::Simple, and its not abundantly clear from *its* documentation wether or not running HTTP::Server::Simple::CGI based scripts is even possible under either a FCGI-slave setup, or underneath something more modern like a PSGI-server.

( Its well documented how to use PSGI underneath HTTP::Server::Simple, but not vice versa )

http://guide.munin-monitoring.org/en/latest/installation/upgrade.html 

Basically says "We're ditching it, so if you we're using it, too bad", and doesn't explain how to use it under an existing HTTP server.

http://guide.munin-monitoring.org/en/latest/example/webserver/apache-virtualhost.html#example-webserver-apache

At best, this article suggests if you want this, you're forced to use a HTTP reverse proxy instead.

Collectively, these sound like consequences of poor design choices, and you might have had a more flexible configuration if it was based on the PSGI standard instead, and having the defacto implementation self-hosting under HTTP::Server::Simple ( which is doable due to HTTP::Server::Simple::PSGI )

**Comment by steveschnepp** [comment]
@kentfredric I agree that CGI is a poor design choice. But that wasn't the case back then when it was made.

I didn't want to go the PSGI road, as I didn't know much about it (more than "it exists"). I therefore (selfishly) choosed the easiest path for me. 

That said : 

* @ssm does a great job at rewriting parts of munin based off Mojo, which offers even more configuration options. 
* I don't think porting the current 2.1 codebase to PSGI would be very complicated for someone with good PSGI knowledge. If it enables then a very simple usage as it currently is, I'm all for merging it.

**Comment by kentfredric** [comment]
> @ssm does a great job at rewriting parts of munin based off Mojo, which offers even more
> configuration options. 

If this is the case, then I have much less to despair about =). Because you can run any Mojo based application under a PSGI server and Mojo will speak PSGI on your behalf, while still offering the convenience of a boxed server.

http://mojolicio.us/perldoc/Mojolicious/Guides/Cookbook#PSGI-Plack

**Comment by steveschnepp** [comment]
Yes, that's exactly what he said this afternoon while I was writing the comment :)
